packages/react-native-reanimated/src/mock.ts

```diff
@@ -50,7 +50,16 @@ const hook = {
   ): EventHandlerProcessed<Event, Context> => NOOP,
   // useHandler: ADD ME IF NEEDED
   useWorkletCallback: ID,
-  useSharedValue: <Value>(init: Value) => ({ value: init }),
+  useSharedValue: <Value>(init: Value) => {
+    let value = init;
+
+    return {
+      set: (newValue: Value) => {
+        value = newValue;
+      },
+      get: () => value
+    };
+  },
```

Comment:
We need to preserve the access by `sv.value`. Try something like this:

```suggestion
  useSharedValue: <Value>(init: Value) => {
    let value = {value: init};
    value.get = () => value.value;
    value.set = (newValue) => { value.value = newValue};
    return value;
  },
```

Or better yet, use a Proxy here.

Reply:
By proxy do you mean something like this?
`
  const useSharedValue = <Value>(init: Value) => {
    let internalValue = init;

    const handler = {
      get(target: any, prop: string) {
        if (prop === 'value') {
          return internalValue;
        }
        if (prop === 'get') {
          return () => internalValue;
        }
        if (prop === 'set') {
          return (newValue: Value) => {
            internalValue = newValue;
          };
        }
        return undefined;
      },
      set(target: any, prop: string, value: any) {
        if (prop === 'value') {
          internalValue = value;
          return true;
        }
        return false;
      }
    };
    return new Proxy({}, handler);
  },
  `

Reply:
I meant the literal [Proxy](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Proxy) object.